src/main/webapp/app/iris/exercise-chatbot/exercise-chatwidget/exercise-chat-widget.component.ts

```diff
@@ -134,11 +136,12 @@ export class ExerciseChatWidgetComponent implements OnInit, OnDestroy, AfterView
     }
 
     ngOnInit() {
-        setTimeout(() => {
-            if (this.messages.length != 0) {
-                this.userAccepted = true;
+        this.userService.getIrisAcceptedAt().subscribe((res) => {
+            this.userAccepted = !!res;
+            if (this.userAccepted && this.messages.length == 0) {
+                this.loadFirstMessage();
```

Comment:
@RY997 by calling to load first message here I think it will solve the problem too

Reply:
The messages should be the value in stateStore which is after this part, so I'm not sure if this works.

Reply:
@RY997 ah then I'll leave it to your PR then. I think we can merge it like this now if there are no bugs

Reply:
I plan to test once and if there's no bugs, we can merge